mlflow/tracking/fluent.py

```diff
@@ -1021,3 +1026,50 @@ def _get_experiment_id():
         or _get_experiment_id_from_env()
         or (is_in_databricks_notebook() and get_notebook_id())
     ) or deprecated_default_exp_id
+
+
+def autolog(log_input_example=False, log_model_signature=True):  # pylint: disable=unused-argument
+    locals_copy = locals().items()
+
+    # Mapping of library module name to specific autolog function
+    # eg: mxnet.gluon is the actual library, mlflow.gluon.autolog is our autolog function for it
+    LIBRARY_TO_AUTOLOG_FN = {
+        "tensorflow": tensorflow.autolog,
+        "keras": keras.autolog,
+        "mxnet.gluon": gluon.autolog,
+        "xgboost": xgboost.autolog,
+        "lightgbm": lightgbm.autolog,
+        "sklearn": sklearn.autolog,
+        "fastai": fastai.autolog,
+    }
+
+    def setup_autologging(module):
+        autolog_fn = LIBRARY_TO_AUTOLOG_FN[module.__name__]
+        try:
+            needed_params = list(inspect.signature(autolog_fn).parameters.keys())
+            filtered = {k: v for k, v in locals_copy if k in needed_params}
+        except ValueError:
+            filtered = {}
+
+        try:
+            autolog_fn(**filtered)
+            _logger.info("Autologging successfully enabled for " + module.__name__ + ".")
+        except Exception as e:
+            _logger.warning(
+                "Exception raised while enabling autologging for " + module.__name__ + ": " + str(e)
+            )
+
+    # for each autolog library (except pyspark), register a post-import hook.
+    # this way, we do not send any errors to the user until we know they are using the library.
+    # the post-import hook also retroactively activates for previously-imported libraries.
+    for module in LIBRARY_TO_AUTOLOG_FN.keys():
+        register_post_import_hook(setup_autologging, module)
+
+    # for pyspark, we activate autologging immediately, without waiting for a module import.
+    # this is because on Databricks a SparkSession already exists and the user can directly
+    #   interact with it, and this activity should be logged.
+    try:
+        spark.autolog()
+        _logger.info("Autologging successfully enabled for spark.")
```

Comment:
There's one more case I hadn't thought about: If PySpark isn't installed, this throws an ImportError. If PySpark is installed, I think we should call `spark.autolog()` immediately; this will handle the Databricks case. Otherwise, we should register an import hook.